addons/forum-search/userscript.js

```diff
@@ -255,21 +266,24 @@ export default async function ({ addon, console, msg }) {
           .getElementsByTagName("li")[2]
           .innerText.substring(2)
           .trim();
-        locationQuery = ` +topic:${pathSplit[3]}`;
+        filterBar.value = `topic.id = ${pathSplit[3]}`;
         searchPlaceholder = msg("search-topic", { topic: topicTitle });
         break;
       }
       case 4: {
         let category = document.getElementsByClassName("box-head")[1].getElementsByTagName("span")[0].textContent;
-        locationQuery = ` +category:"${category}"`;
+        filterBar.value = ` topic.category = "${category}"`;
         searchPlaceholder = msg("search-cat", { cat: category });
         break;
       }
     }
   }
 
   searchBar.setAttribute("placeholder", searchPlaceholder);
+  filterBar.setAttribute("placeholder", msg("filter-placeholder"));
   search.appendChild(searchBar);
+  search.appendChild(filterBar);
+  search.appendChild(Object.assign(document.createElement("input"), { type: "submit", style: "display: none" })); // this is a hack to make the enter key work
```

Comment:
I actually think the button should be visible and labelled "Search" - it's weird that there currently isn't one.

Reply:
You can now remove the invisible button.